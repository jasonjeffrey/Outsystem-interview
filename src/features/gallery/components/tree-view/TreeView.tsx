import React, {FunctionComponent} from 'react';
import {List} from '@material-ui/core';
import {TreeViewItem} from '../tree-view-item/TreeViewItem';
import {IFolder} from '../../interfaces/i-folder';

type TreeViewProps = {
    folders: IFolder[];
}

export const TreeView: FunctionComponent<TreeViewProps> = ({folders}) => {
    return (
    <List>
        {folders.map((folder: IFolder) => (<TreeViewItem key={folder.name} title={folder.name} artists={folder.artists}/>))}
    </List>
)};
