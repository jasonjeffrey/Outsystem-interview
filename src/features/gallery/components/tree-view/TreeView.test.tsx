import {render} from '@testing-library/react';
import {TreeView} from './TreeView';

jest.mock('react-redux', () => ({
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

describe('TreeView Component', () => {
    it('RENDERS component without any TreeViewItems', () => {
        const component = <TreeView folders={[]}/>
        const {asFragment} = render(component);

        expect(asFragment()).toMatchSnapshot();
    });

    it('RENDERS component with 1 TreeViewItem', () => {
        const component = <TreeView folders={[{name: 'test', artists: []}]}/>
        const {asFragment} = render(component);

        expect(asFragment()).toMatchSnapshot();
    });
});
