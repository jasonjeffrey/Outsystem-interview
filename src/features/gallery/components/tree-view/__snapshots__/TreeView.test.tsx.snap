// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreeView Component RENDERS component with 1 TreeViewItem 1`] = `
<DocumentFragment>
  <ul
    class="MuiList-root MuiList-padding"
  >
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      data-testid="test"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          test
        </span>
      </div>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
  </ul>
</DocumentFragment>
`;

exports[`TreeView Component RENDERS component without any TreeViewItems 1`] = `
<DocumentFragment>
  <ul
    class="MuiList-root MuiList-padding"
  />
</DocumentFragment>
`;
